tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -76,6 +66,12 @@ public:
 
    void AddFriend(const std::vector<std::pair<std::string, std::string>> &treeAndFileNameGlobs,
                   const std::string &alias = "");
+
+   const std::vector<std::string> &GetTreeNames() const;
+   const std::vector<std::string> &GetFileNameGlobs() const;
+   const Long64_t &GetEntryRangeBegin() const;
+   const Long64_t &GetEntryRangeEnd() const;
+   const ROOT::Internal::TreeUtils::RFriendInfo &GetFriendInfo() const;
```

Comment:
`RDatasetSpec` is user-facing, so now `RFriendInfo` is user-facing too (which is not a bad thing). Can we take it out of `Internal`? cc: @vepadulano

Reply:
Probably, if it goes under the same changes of `RDatasetSpec` to become a "read-only" class.

Reply:
I don't understand the if condition :) even before this patch, users handle RDatasetSpec objects. With this patch, those objects will have a `GetFriendInfo` method that would return an `Internal` data structure, but it would make more sense to return to users types that we say are meant to be handled by users (so not `Internal`).

Reply:
As a follow-up on @vepadulano 's comment - currently `RFriendInfo` is a struct with public members only, currently used as a "builder". We can instead another class/struct for friendInfo which would be "read-only".

I would suggest to return `ROOT::Internal::TreeUtils::RFriendInfo` for now, and once we have the spec builder, we also introduce analogous pattern for the friend info.

Reply:
Yes I mean that `RFriendInfo` should become a `class` and only present getters to the users, like `RDatasetSpec` is becoming. It is true that this PR introduces a getter that returns an internal class, but it's a temporary state that needs to be changed soon in a subsequent PR (in any case because the `RDatasetSpec` will be created only by the builder)

Reply:
> We can instead another class/struct for friendInfo which would be "read-only".

It's very easy to make `RFriendInfo` read-only: it's enough to return a `const RFriendInfo &` from the getter in RDatasetSpec :smile: 

> once we have the spec builder, we also introduce analogous pattern for the friend info.

this is a separate discussion, independent of the namespace in which `RFriendInfo` should live.

> it's a temporary state that needs to be changed soon in a subsequent PR (in any case because the RDatasetSpec will be created only by the builder)

I don't see why adding a builder for RDatasetSpec will require changing the signature of its getters.

All in all I don't see why we cannot extract RFriendInfo from the Internal namespace already at this time, but as long as it's done for 6.28 it's effectively the same thing, so as you prefer :)